### ISSUE TITLE
rclone removing excluded files when 'pulling' the sync

### ISSUE THREAD

**Issue by enkia** [ncw/rclone#205]
I set up rclone on my main computer to sync a folder to Amazon Cloud Drive:
```
rclone sync --checksum --bwlimit=400k --filter-from="rclone-filters.txt" ~/sites acd:sites
```
And another on my laptop to 'pull' the folder:
```
rclone sync --checksum --bwlimit=1M --filter-from="rclone-filters.txt" acd:sites ~/sites
```

rclone-filters.txt excludes .gir directories, node_modules, etc so the version on ACD doesn't have these folders.

When syncing to the laptop, rclone deletes these directories on the laptop if they already existed. (Or tries to in the case of .git directories which just spams the terminal with operation not permitted errors) Shouldn't using --filter-from on the 'pull' command prevent that?

**Comment by ncw** [comment]
I seem what you mean.  Here is my reproduction

```
$ mkdir a
$ mkdir b
$ touch a/one.jpg
$ rclone -v sync --exclude "*.jpg" b a
2015/11/09 07:40:50 Local file system at /tmp/a: Modify window is 1ns
2015/11/09 07:40:50 Local file system at /tmp/a: Building file list
2015/11/09 07:40:50 Local file system at /tmp/a: Waiting for checks to finish
2015/11/09 07:40:50 Local file system at /tmp/a: Waiting for transfers to finish
2015/11/09 07:40:50 Waiting for deletions to finish
2015/11/09 07:40:50 one.jpg: Deleted
```

`one.jpg` shouldn't have been deleted unless `--delete-excluded` was passed in.

Can you tell me make me a reproduction like the above which shows rclone trying to delete .git directories and spamming the terminal?  I haven't been able to do that.

Thanks

**Comment by enkia** [comment]
Like this? This is only a very very small snippet of the spam: 

```shell
2015/11/09 09:36:15 testsite.dev/.git/objects/68/5bf84b5795bd05d8e9f3ca1ba1067ee872de06: Couldn't delete: remove /Users/enkia/sites/testsite.dev/.git/objects/68/5bf84b5795bd05d8e9f3ca1ba1067ee872de06: operation not permitted
2015/11/09 09:36:15 testsite.dev/.git/objects/38/424b80ac6bb5183e3cb608af0cbf07f849e1fc: Couldn't delete: remove /Users/enkia/sites/testsite.dev/.git/objects/38/424b80ac6bb5183e3cb608af0cbf07f849e1fc: operation not permitted
2015/11/09 09:36:15 craft.dev/.git/objects/6e/52393086d16c538bdff337a3d34580db470d9f: Couldn't delete: remove /Users/enkia/sites/craft.dev/.git/objects/6e/52393086d16c538bdff337a3d34580db470d9f: operation not permitted
2015/11/09 09:36:15 testsite.dev/.git/objects/41/323441a22cea15c56c7061709e3f9c821538fd: Couldn't delete: remove /Users/enkia/sites/testsite.dev/.git/objects/41/323441a22cea15c56c7061709e3f9c821538fd: operation not permitted
2015/11/09 09:36:15 testsite.dev/.git/objects/8f/8375c88d16c87d04039dad8b8c5abe07cc3e1e: Couldn't delete: remove /Users/enkia/sites/testsite.dev/.git/objects/8f/8375c88d16c87d04039dad8b8c5abe07cc3e1e: operation not permitted
2015/11/09 09:36:15 anothertestsite/.git/objects/6a/94c9d34de68039883dee0a620e5d45ee89774f: Couldn't delete: remove /Users/enkia/sites/anothertestsite/.git/objects/6a/94c9d34de68039883dee0a620e5d45ee89774f: operation not permitted
2015/11/09 09:36:15 testsite.dev/.git/objects/a7/0a51b1fcee3ac128dfad334bf458d122debadb: Couldn't delete: remove /Users/enkia/sites/testsite.dev/.git/objects/a7/0a51b1fcee3ac128dfad334bf458d122debadb: operation not permitted
2015/11/09 09:36:15 craft.dev/.git/objects/5c/94977741825bf61d0dbf7314ac61978385b0e7: Couldn't delete: remove /Users/enkia/sites/craft.dev/.git/objects/5c/94977741825bf61d0dbf7314ac61978385b0e7: operation not permitted
2015/11/09 09:36:15 testsite.dev/.git/objects/4f/0bbd446af64cd8ed9e84fd2d779d7ba0b5b4a3: Couldn't delete: remove /Users/enkia/sites/testsite.dev/.git/objects/4f/0bbd446af64cd8ed9e84fd2d779d7ba0b5b4a3: operation not permitted
2015/11/09 09:36:15 testsite.dev/.git/objects/20/abcba3e8269da28fcf2fe374d470a06dfc5a90: Couldn't delete: remove /Users/enkia/sites/testsite.dev/.git/objects/20/abcba3e8269da28fcf2fe374d470a06dfc5a90: operation not permitted
2015/11/09 09:36:15 testsite.dev/.git/objects/b0/f91e22c9b5e5ba22abc104a7d7c19ec7cacfef: Couldn't delete: remove /Users/enkia/sites/testsite.dev/.git/objects/b0/f91e22c9b5e5ba22abc104a7d7c19ec7cacfef: operation not permitted
2015/11/09 09:36:15 craft.dev/.git/objects/6c/ccbbbfc3479909db888e48e75ec14f21aeeda5: Couldn't delete: remove /Users/enkia/sites/craft.dev/.git/objects/6c/ccbbbfc3479909db888e48e75ec14f21aeeda5: operation not permitted
2015/11/09 09:36:15 testsite.dev/.git/objects/6b/79597ca3536290a42287fd6fb26a2195d3bfea: Couldn't delete: remove /Users/enkia/sites/testsite.dev/.git/objects/6b/79597ca3536290a42287fd6fb26a2195d3bfea: operation not permitted
2015/11/09 09:36:15 anothertestsite/.git/objects/7e/7c0be4225797732cf20f426ae8120600b1783d: Couldn't delete: remove /Users/enkia/sites/anothertestsite/.git/objects/7e/7c0be4225797732cf20f426ae8120600b1783d: operation not permitted
2015/11/09 09:36:15 testsite.dev/.git/objects/01/a47870194b27b792e5a0c3c9de254d697a8603: Couldn't delete: remove /Users/enkia/sites/testsite.dev/.git/objects/01/a47870194b27b792e5a0c3c9de254d697a8603: operation not permitted
2015/11/09 09:36:15 craft.dev/.git/objects/13/f055a8224f69fdaca487bafa9fbea004e05c12: Couldn't delete: remove /Users/enkia/sites/craft.dev/.git/objects/13/f055a8224f69fdaca487bafa9fbea004e05c12: operation not permitted
2015/11/09 09:36:15 testsite.dev/.git/objects/08/93d3ad1be6246188ff6b574b9635c614ce15ff: Couldn't delete: remove /Users/enkia/sites/testsite.dev/.git/objects/08/93d3ad1be6246188ff6b574b9635c614ce15ff: operation not permitted
2015/11/09 09:36:15 testsite.dev/.git/objects/f2/05e166c7dcaad0dd71f978b5dbff678850f36e: Couldn't delete: remove /Users/enkia/sites/testsite.dev/.git/objects/f2/05e166c7dcaad0dd71f978b5dbff678850f36e: operation not permitted
2015/11/09 09:36:15 craft.dev/.git/objects/05/1f27f2e9ef0a75deff2b46d3474653a79e6346: Couldn't delete: remove /Users/enkia/sites/craft.dev/.git/objects/05/1f27f2e9ef0a75deff2b46d3474653a79e6346: operation not permitted
2015/11/09 09:36:15 testsite.dev/.git/objects/bf/93ed5933d536297301ad3c7c3f2323fa24f5a0: Couldn't delete: remove /Users/enkia/sites/testsite.dev/.git/objects/bf/93ed5933d536297301ad3c7c3f2323fa24f5a0: operation not permitted
2015/11/09 09:36:15 Attempt 1/3 failed with 12792 errors
2015/11/09 09:36:26 Local file system at /Users/enkia/sites: Waiting for checks to finish
2015/11/09 09:37:02
Transferred:         1060 Bytes (   0.00 kByte/s)
Errors:                 0
Checks:             80696
Transferred:           11
Elapsed time:     12m1.2s

2015/11/09 09:38:02
Transferred:         1060 Bytes (   0.00 kByte/s)
Errors:                 0
Checks:             92312
Transferred:           11
Elapsed time:     13m1.2s

2015/11/09 09:39:02
Transferred:         1060 Bytes (   0.00 kByte/s)
Errors:                 0
Checks:            101214
Transferred:           11
Elapsed time:     14m1.1s
```

**Comment by ncw** [comment]
Thanks.  Can you tell me what you've got in your  `rclone-filters.txt` file and I'll try to reproduce that problem too.

**Comment by enkia** [comment]
```
- _gsdata_/**
- _saved_/**
- node_modules/**
- bower_components/**
- **.git/**
- .DS_Store
- Thumbs.db
- .vagrant/**
- .sass-cache/**
- **.bower**
- .debris/**
- *.com*
- Icon?
- .otto/**
- files/cache/**
- files/tmp/**
+ .gitignore
```

**Comment by ncw** [comment]
Thanks for doing that.  I think the `.git` spam is just because you don't have permission to delete those files rather than a problem with rclone.

It will go away when I fix the underlying issue.

**Comment by enkia** [comment]
My issue is that I don't want rclone to delete these excluded directories.  It also deleted node_modules, bower_components, etc.  It shouldn't even know they exist if using the --filter-from option, right?

To reiterate, I uploaded the directory using the `--filter-from` option on the **first computer:**
```
rclone sync --checksum --bwlimit=400k --filter-from="rclone-filters.txt" **~/sites** acd:sites
```

Then on **another computer**, which already had the sites directory but an outdated one, I used the below command to pull/update the sites directoryr:

```
rclone sync --checksum --bwlimit=1M --filter-from="rclone-filters.txt" acd:sites **~/sites**
```

It then proceeded to delete all the directories on the second computer that the remote version didn't have despite the inclusion of the `--filter-from` command. 

Perhaps it's just because I'm doing it wrong? How can I get rclone to completely ignore directories?
